EKF/control.cpp

```diff
@@ -740,6 +737,16 @@ void Ekf::controlHeightFusion()
 			if (_control_status_prev.flags.baro_hgt != _control_status.flags.baro_hgt) {
 				_hgt_sensor_offset = 0.0f;
 			}
+		} else if (_control_status_prev.flags.gps_hgt && _gps_data_ready && !_gps_hgt_faulty) {
+			setControlGPSHeight();
+			_fuse_height = true;
+			_in_range_aid_mode = false;
+
+			// we have just switched to using gps height, calculate height sensor offset such that current
+			// measurment matches our current height estimate
+			if (_control_status_prev.flags.gps_hgt != _control_status.flags.gps_hgt) {
```

Comment:
In that case, the call to `setControlGPSHeight()` isn't strictly necessary, right?